docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+        "another_unique_dataset_identifier",
+        dfs=[{"data": {"table1": df, "table2": df2}}],
+    )
+    dm.tables
+    sorted(dm.partitions.keys())
+
+
+.. admonition:: Filenames
+
+   In the previous example a dictionary was used to pass the desired data to the store function. To label each file,
+   by default ``kartothek`` uses UUIDs to ensure that each filename is unique, this is necessary so that the
+   update can properly work using `copy-on-write <https://en.wikipedia.org/wiki/Copy-on-write>`_ principles.
+
+Below is an example where we update the existing dataset ``another_unique_dataset_identifier``
+with new data for ``table1`` and ``table2``:
+
+.. ipython:: python
+
+    another_df2 = pd.DataFrame(
+        {
+            "G": "bar",
+            "H": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "I": np.array([12] * 4, dtype="int32"),
+            "J": pd.Series(4, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190614"),
+            "L": 10.0,
+        }
+    )
+    another_df2
+
+    dm = update_dataset_from_dataframes(
+        {"data": {"table1": another_df, "table2": another_df2}},
+        store=store_factory,
+        dataset_uuid="another_unique_dataset_identifier",
+    )
+    dm.tables
+    sorted(dm.partitions.keys())
+
+
+Trying to update only a subset of tables throws a ``ValueError``:
+
+.. ipython::
+
+    @verbatim
+    In [45]: update_dataset_from_dataframes(
+       ....:        {
+       ....:           "data":
+       ....:           {
+       ....:              "table2": another_df2
+       ....:           }
+       ....:        },
+       ....:        store=store_factory,
+       ....:        dataset_uuid="another_unique_dataset_identifier"
+       ....:        )
+       ....:
+    ---------------------------------------------------------------------------
+    ValueError: Input partitions for update have different tables than dataset:
+    Input partition tables: {'table2'}
+    Tables of existing dataset: ['table1', 'table2']
+
+
+Adding data to an existing dataset is not the only functionality achievable within an update
+operation, such an operation can also be used to remove or overwrite data.
+To do so we use the ``delete_scope`` keyword argument as shown in the example below:
+
+.. ipython:: python
+
+    dm = update_dataset_from_dataframes(
+        None,
+        store=store_factory,
+        dataset_uuid="partitioned_dataset",
+        partition_on="E",
+        delete_scope=[{"E": "train"}],
+    )
+    sorted(dm.partitions.keys())
+
+
+As we can see, we specified using a dictionary that data where the column ``E`` has the
+value ``train`` should be removed. Looking at the partitions after the update, we see that
+the partition ``E=train`` has been removed.
+
+.. note:: We defined ``delete_scope`` over a value of ``E`` which is also the column that
+    we partitioned on: ``delete_scope`` only works *as expected* on indexed columns.
```

Comment:
It *only* works on indexed columns and the recommendation is to use it for *partition columns* only

Reply:
Feel free to promote this from `note` to `warning`

Reply:
> It _only_ works on indexed columns and the recommendation is to use it for _partition columns_ only

The recommendation is in the next line in the (now) warning.